opslevel/resource_opslevel_check_package_version.go

```diff
@@ -0,0 +1,352 @@
+	"github.com/opslevel/opslevel-go/v2024"
+	"github.com/relvacode/iso8601"
+)
+
+var (
+	_ resource.ResourceWithConfigure   = &CheckPackageVersionResource{}
+	_ resource.ResourceWithImportState = &CheckPackageVersionResource{}
+)
+
+func NewCheckPackageVersionResource() resource.Resource {
+	return &CheckPackageVersionResource{}
+}
+
+// CheckPackageVersionResource defines the resource implementation.
+type CheckPackageVersionResource struct {
+	CommonResourceClient
+}
+
+type CheckPackageVersionResourceModel struct {
+	Category    types.String `tfsdk:"category"`
+	Description types.String `tfsdk:"description"`
+	Enabled     types.Bool   `tfsdk:"enabled"`
+	EnableOn    types.String `tfsdk:"enable_on"`
+	Filter      types.String `tfsdk:"filter"`
+	Id          types.String `tfsdk:"id"`
+	Level       types.String `tfsdk:"level"`
+	Name        types.String `tfsdk:"name"`
+	Notes       types.String `tfsdk:"notes"`
+	Owner       types.String `tfsdk:"owner"`
+
+	MissingPackageResult       types.String `tfsdk:"missing_package_result"`
+	PackageConstraint          types.String `tfsdk:"package_constraint"`
+	PackageManager             types.String `tfsdk:"package_manager"`
+	PackageName                types.String `tfsdk:"package_name"`
+	PackageNameIsRegex         types.Bool   `tfsdk:"package_name_is_regex"`
+	VersionConstraintPredicate types.Object `tfsdk:"version_constraint_predicate"`
+}
+
+func ParsePredicate(predicate *opslevel.Predicate) types.Object {
+	if predicate == nil {
+		return types.ObjectNull(predicateType)
+	}
+	predicateAttrValues := map[string]attr.Value{
+		"type":  types.StringValue(string(predicate.Type)),
+		"value": OptionalStringValue(predicate.Value),
+	}
+	return types.ObjectValueMust(predicateType, predicateAttrValues)
+}
+
+func NewCheckPackageVersionResourceModel(ctx context.Context, check opslevel.Check, planModel CheckPackageVersionResourceModel) CheckPackageVersionResourceModel {
+	var stateModel CheckPackageVersionResourceModel
+
+	stateModel.Category = RequiredStringValue(string(check.Category.Id))
+	if planModel.Enabled.IsNull() {
+		stateModel.Enabled = types.BoolValue(false)
+	} else {
+		stateModel.Enabled = OptionalBoolValue(&check.Enabled)
+	}
+	if planModel.EnableOn.IsNull() {
+		stateModel.EnableOn = types.StringNull()
+	} else {
+		// We pass through the plan value because of time formatting issue to ensure the state gets the exact value the customer specified
+		stateModel.EnableOn = planModel.EnableOn
+	}
+	stateModel.Filter = OptionalStringValue(string(check.Filter.Id))
+	stateModel.Id = ComputedStringValue(string(check.Id))
+	stateModel.Level = RequiredStringValue(string(check.Level.Id))
+	stateModel.Name = RequiredStringValue(check.Name)
+	stateModel.Notes = OptionalStringValue(check.Notes)
+	stateModel.Owner = OptionalStringValue(string(check.Owner.Team.Id))
+
+	if check.MissingPackageResult != nil {
+		stateModel.MissingPackageResult = RequiredStringValue(string(*check.MissingPackageResult))
```

Comment:
```suggestion
		stateModel.MissingPackageResult = OptionalStringValue(string(*check.MissingPackageResult))
```
Optional in the schema and the OneOf(...) validator ensures this won't be an empty string